### DISCUSSION TITLE
Middleware for InputType fields

### DISCUSSION THREAD

**Discussion by skynightz93** [MichalLytek/type-graphql#913] [General]
Hi, 

Can I confirm either field in @InputType can have its own middleware?

My use case is to change the value of @InputType if it has its own middleware. For example, I called it as `ChangeToLike` middleware. This middleware will automatically change the value to TypeORM "Like" statement before passing the value to Resolver.

My current way of doing things is to change the value in Resolver which is very troublesome for each of the fields that need to be changed to "Like" statement. Or, do you have any other suggestions for this use case?

**Comment by MichalLytek** [comment]
No, input types have no resolvers, thus no middlewares.

**Comment by skynightz93** [comment]
For those seeking the same feature, the best way now is to create a function that does this feature in the resolver functions. For example, I am using this kind of function for now which selects fields name in the array, then it will convert to Like.

```
where = ChangeToLike(where, ["DisplayName", "Email", "GivenName", "FamilyName"]);
```
